src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1321,21 +1368,15 @@ public SolutionCompilationState WithoutFrozenSourceGeneratedDocuments()
     /// generated file open, we need to make sure everything lines up.
     /// </summary>
     public SolutionCompilationState WithFrozenSourceGeneratedDocuments(
-        ImmutableArray<(SourceGeneratedDocumentIdentity documentIdentity, DateTime generationDateTime, SourceText sourceText)> documents)
+        ImmutableArray<(SourceGeneratedDocumentIdentity documentIdentity, DateTime generationDateTime, SourceText sourceText, SyntaxNode? syntaxNode)> documents)
```

Comment:
I don't like the options syntax node any more than you do, but I couldn't work out a nice way to allow this without just duplicating all of the code in this method, with another set of methods, or creating a method with a pretty ugly signature separate delegates for create and update etc. Open to suggestions.

Reply:
given that you can do WithSyntaxRoot now on an SG doc, this makes some amount fo sense to me.